### PR TITLE
Add Syncthing integration

### PR DESCRIPTION
**Pull request by sashacmc**
### What does this PR do?

Add [Syncthing](https://syncthing.net/) integration

### Motivation

Connect the Syncthing sync software to the Datadog 

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by alai97** [comment]
Hey @sashacmc, do we plan on documenting the rest of the metrics in the metadata.csv file? If so, I will create an editorial review card and someone from Docs will do a complete review when this integration documentation is ready to go 😛

**Comment by sashacmc** [comment]
> Hey @sashacmc, do we plan on documenting the rest of the metrics in the metadata.csv file? If so, I will create an editorial review card (DOCS-3327) and someone from Docs will do a complete review when this integration documentation is ready to go 😛

Done!

File: syncthing/tests/test_syncthing.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+
+
+# def test_emits_critical_service_check_when_service_is_down(dd_run_check, aggregator, instance):
+#    # type: (Callable[[AgentCheck, bool], None], AggregatorStub, Dict[str, Any]) -> None
+#    check = SyncthingCheck('syncthing', {}, [instance])
+#    dd_run_check(check)
+#   # aggregator.assert_service_check('syncthing.can_connect', SyncthingCheck.CRITICAL)
```
Comment (hithwen) [inline syncthing/tests/test_syncthing.py:174]: Why is this commented out?

File: syncthing/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Agent Check: Syncthing
+
+## Overview
+
+This check monitors [Syncthing][1].
```
Comment (urseberry) [inline syncthing/README.md:5]: Can you provide more detail here? Something like:

"Syncthing synchronizes files between two or more computers in real time. This integration allows you to monitor [Syncthing][1] using Datadog."

File: syncthing/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Setup
+
+The Syncthing check is not included in the [Datadog Agent][2] package, so you need to install it.
+
+### Installation
+
+For Agent v7.21+ / v6.21+, follow the instructions below to install the Syncthing check on your host. See [Use Community Integrations][3] to install with the Docker Agent or earlier versions of the Agent.
```
Comment (urseberry) [inline syncthing/README.md:13]:
```suggestion
For Agent v7.21+ / v6.21+, follow the instructions below to install the Syncthing check on your host. See [Use Community Integrations][3] to install the Syncthing check with the Docker Agent or earlier versions of the Agent.
```

**Review by urseberry — APPROVED** [review]

**Review by urseberry — COMMENTED** [review]
The metadata.csv file needs fixing. The fields are out of order, and some are missing. For example, line 2 is: ```syncthing.connections.connected,gauge,,connection,,,0,syncthing,The number of connected endpoints,```

The phrase "The number of connected enpoints" is currently in the `short_name` column. Instead, it should go under `description`. And then you need a different `short_name` for each metric.

File: syncthing/README.md (hunk 2 above)
Comment (sashacmc) [reply syncthing/README.md]: Fixed!

**Review by urseberry — APPROVED** [review]

**Review by ofek — APPROVED** [review]
Looks great, thanks!

**Pull request merged**